sdk/monitor/monitor-opentelemetry/test/internal/unit/main.test.ts

```diff
@@ -251,4 +251,70 @@ describe("Main functions", () => {
     useAzureMonitor(config);
     assert.strictEqual(process.env["AZURE_MONITOR_PREFIX"], `k${os}m_`);
   });
+
+  it("should prioritize resource detectors in env var OTEL_NODE_RESOURCE_DETECTORS", () => {
+    const expectedResourceAttributeNamespaces =
+      new Set(["os", "service", "telemetry"]);
+    const env = <{ [id: string]: string }>{};
+    env.OTEL_NODE_RESOURCE_DETECTORS = "os";
+    process.env = env;
+    let config: AzureMonitorOpenTelemetryOptions = {
+      azureMonitorExporterOptions: {
+        connectionString: "InstrumentationKey=00000000-0000-0000-0000-000000000000",
+      },
+    };
+    useAzureMonitor(config);
+    let span = trace.getTracer("testTracer").startSpan("testSpan");
+    span.end();
+    //@ts-ignore
```

Comment:
If including a `ts-ignore`, you should add some comment here to explain why it's necessary and typing isn't allowing you to achieve what you want in this test.